Диплом. Часть 2.

AAsadova34 opened the pull request.

AAsadova34 committed:
Диплом. Часть 2.

AAsadova34 committed:
Диплом. Часть 2. Ревью 1.
Рабочая локально версия.

EvgenyKoptubenko commented:
Если удалить не существующего пользователя, то падает
{
	"status": "INTERNAL_SERVER_ERROR",
	"reason": "Unknown reason",
	"message": "An unexpected error has occurred",
	"timestamp": "2023-03-06 20:55:35"
}

curl --request DELETE \
  --url http://localhost:8080/admin/users/12 \
  --header 'Content-Type: application/json' \
  --data '{
	"name":"nameItem",
	"email":"email3email.com",
  "login":"login",
	"birthday":"2012-04-23T18:25:43.511Z"
}'

2023-03-06 20:55:35.549  WARN 1 --- [nio-8080-exec-1] ru.practicum.explore.ewm.utility.Logger  : EmptyResultDataAccessException
ewm_service_container   | 
ewm_service_container   | org.springframework.dao.EmptyResultDataAccessException: No class ru.practicum.explore.ewm.user.User entity with id 12 exists!

EvgenyKoptubenko commented:
Приложение позволяет добавить пользователя с неверным email
curl --request POST \
  --url http://localhost:8080/admin/users \
  --header 'Content-Type: application/json' \
  --data '{
	"name":"namessssItem",
	"email":"email3email.com",
  "login":"login",
	"birthday":"2012-04-23T18:25:43.511Z"
}'

EvgenyKoptubenko reviewed (changes requested):
Работа отличная. Поправь пару замечаний и присылай

AAsadova34 committed:
Диплом. Часть 2. Ревью 2.
fix: изменить класс ошибки, отлавливаемой, при вызове метода deleteById()  на EmptyResultDataAccessException
fix: добавить проверку на валидность email в UserDto

EvgenyKoptubenko reviewed (approved):
Работа сделана хорошо.
